base_decorator.rb

```diff
@@ -0,0 +1,14 @@
+require './nameable'
+
+class BaseDecorator < Nameable
+  # In the constructor assign a nameable object from params to an instance variable.
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  #   Implement the correct_name method that returns the result of the correct_name method of the @nameable.
+  def correct_name
+    @nameable.correct_name
+  end
+end
```

Comment:
- [ ] Please make sure the `nameable` variable is readable and writable to work the decorator pattern as expected.
  - You can consider using `attr_accessor :nameable`